File: android/app/src/test/java/com/shadelight/iausage/data/ProviderVisualsTest.kt
package com.shadelight.iausage.data

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Assert.assertNull
import org.junit.Test

class ProviderVisualsTest {
    @Test fun `known providers resolve a real icon`() {
        assertNotNull(providerVisual("anthropic", "Claude Code").icon)
        assertNotNull(providerVisual("openai", "Codex").icon)
        assertNotNull(providerVisual("antigravity", "Antigravity").icon)
    }

    @Test fun `an unknown provider never breaks - generic fallback plus derived code`() {
        val visual = providerVisual("some-new-vendor", "Some New Vendor")
        assertNull(visual.icon)
        assertEquals("SOM", visual.shortCode)
    }

    @Test fun `a name with no alphanumerics still yields a usable fallback code`() {
        assertEquals("AI", providerVisual("mystery", "!!!").shortCode)
    }

    @Test fun `known providers use the desktop tab codes`() {
        assertEquals("CLD", providerVisual("anthropic", "Claude Code").shortCode)
        assertEquals("CDX", providerVisual("openai", "Codex / ChatGPT").shortCode)
    }

    @Test fun `short names stay compact for small widgets`() {
        assertEquals("Claude", providerShortName("anthropic", "Claude Code"))
        assertEquals("Codex", providerShortName("openai", "Codex / ChatGPT"))
        assertEquals("Grok", providerShortName("grok", "Grok (xAI)"))
        assertEquals("Cursor", providerShortName("cursor", "Cursor"))
    }
}

File: apps/vscode-extension/src/status/provider-visuals.test.ts
import assert from "node:assert/strict";
import { test } from "node:test";
import { providerVisual } from "./provider-visuals";

test("known providers map to their brand icon and short code", () => {
  assert.deepEqual(providerVisual("anthropic", "Claude Code"), { icon: "ia-anthropic", short: "CLD" });
  assert.deepEqual(providerVisual("openai", "Codex"), { icon: "ia-openai", short: "CDX" });
  assert.deepEqual(providerVisual("antigravity", "Antigravity"), { icon: "ia-antigravity", short: "AGY" });
  assert.deepEqual(providerVisual("opencode_go", "OpenCode Go"), { icon: "ia-opencode_go", short: "OCG" });
});

test("an unknown provider never breaks: generic icon + derived short code", () => {
  const visual = providerVisual("some-new-vendor", "Some New Vendor");
  assert.equal(visual.icon, "circle-filled");
  assert.equal(visual.short, "SOM");
});

test("a name with no alphanumerics still produces a 3-letter fallback code", () => {
  const visual = providerVisual("mystery", "!!!");
  assert.equal(visual.icon, "circle-filled");
  assert.equal(visual.short, "AI");
});
